keepvariable/keepvariable_core.py

```diff
@@ -368,10 +378,8 @@ def query(
         :rtype: list[tuple]
         """
         found_jobs: dict[str, dict] = {
```

Comment:
if `tag_params` or `text_params` is not None then you do this twice, what if both `tag_params` and `text_params` are specified?

Reply:
@ZaiZu2 please explain + maybe add assert if at least one of the two needs to always be None (but maybe even better would be to unify it)

Reply:
`query()` method is an in-memory equivalent of Redis `search()` method. In short, Redis allows you to index TEXT/TAG/... fields inside of your saved JSON documents, and use these indexes later on to perform quick searches on these fields. This method mimics this behavior, except it does not need an index to perform the search. To stay compatible with how RediSearch works, it still uses the same concepts, i.e. searching on TEXT/TAG fields. 

We can search through saved documents with any combination of TAG/TEXT params - both can be `None`, then we find all the documents under a given namespace, or both can specify some values to filter on - TEXT params will match any specified glob-pattern string in the field, while TAG params will filter on matching values for the specified fields.